ledger-core/v2/virtual/request/vstatereport.go

```diff
@@ -51,19 +51,18 @@ func (s *SMVStateReport) stepProcess(ctx smachine.ExecutionContext) smachine.Sta
 	objectRef := incomingObjectState.Reference
 	sharedObjectState := catalog.GetOrCreate(ctx, objectRef, object.InitReasonVStateReport)
 
-	setStateFunc := func(data interface{}) (wakeup bool) {
-		state := data.(*object.SharedState)
-		if state.Descriptor() == nil {
+	setStateFunc := func(state *object.SharedState) {
+		if !state.IsReady() {
```

Comment:
Not sure that'll work correctly with "Inactive" object, if we'll receive "Missing" or other state.

Reply:
just sort state:
```
type State int32

const (
	NotReady State = iota
	Missing = 1
	HasState  = 2
	Inactive = 3
)
```

And use ">" for comparing applicable jumps, I guess

Reply:
O_O

Reply:
I think its not good idea to rely on order of these constants. It might be changed accidentally. 
But yes,  question about different states in VStateUnavailable and in SMObject is opened.
Right now it's not clear what to do

Reply:
Relying on order of constants is a common practice, that's why one shouldn't change order of constants without giving it a second thought.

Reply:
Create task for resolving this issue